src/GHC/Util.hs

```diff
@@ -232,3 +233,6 @@ declName (SigD _ (ClassOpSig _ _ (x:_) _)) = occNameString $ occName $ unLoc x
 declName (ForD _ ForeignImport{fd_name}) = occNameString $ occName $ unLoc fd_name
 declName (ForD _ ForeignExport{fd_name}) = occNameString $ occName $ unLoc fd_name
 declName _ = ""
+
+unsafePrettyPrint :: (Outputable.Outputable a) => a -> String
```

Comment:
Moved this here because it often seems useful. Not sure if it's a good idea, as it *is* "unsafe".

Reply:
Would be good to know what the invariants are on it. Does it work after some step in the process? Is that step guaranteed? I'm not sure if `unsafePrettyPrint` is the right name until we've clarified if the unsafety is something we care about.

Reply:
I'm not sure about this one.

Looking at the docs it uses [`unsafeGlobalDynFlags`](http://hackage.haskell.org/package/ghc-lib-parser-8.8.0.20190424/docs/src/DynFlags.html#v_unsafeGlobalDynFlags):
```
-- Do not use unsafeGlobalDynFlags!
--
-- unsafeGlobalDynFlags is a hack, necessary because we need to be able
-- to show SDocs when tracing, but we don't always have DynFlags
-- available.
--
-- Do not use it if you can help it. You may get the wrong value, or this
-- panic!
```

To me it seems that the things in there shouldn't affect us in this case, however I really don't know enough about `ghc` to make this call.
I have the following two ideas in mind if this needs changing:
1. Can make this use `GHC.Util.baseDynFlags`.
2. Use `GHC.Util.parsePragmasIntoDynFlags` and pass the flags down in some form to the pretty printer.

If you're fine with this, let's fix this in another issue/pr - it works for now, and I see no benefit to doing both things at once. (I will add a TODO-like comment to this though)

Reply:
One argument for us attempting to arrange for "safe" pretty prints with the dyn flags used for parsing is that some pretty-prints need to take extensions into account e.g. `ImportQualifiedPost`.

Reply:
Do we expect to actually honour that? My guess is no, even if the extension is set, since it adds a lot of complexity for relatively minor benefit.

Reply:
Nah, I'd be casual about that. If the situation changes significantly we can revisit.